lib/OpenQA/WebAPI/Plugin/AMQP.pm

```diff
@@ -0,0 +1,188 @@
+# Copyright (C) 2016 SUSE Linux LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+package OpenQA::WebAPI::Plugin::AMQP;
+
+use strict;
+use warnings;
+
+use parent qw/Mojolicious::Plugin/;
+use JSON;
+use Mojo::IOLoop;
+use OpenQA::Utils;
+use OpenQA::Schema::Result::Jobs;
+use Mojo::RabbitMQ::Client;
+
+my @job_events     = qw(job_create job_delete job_cancel job_duplicate job_restart job_update_result job_done);
+my @comment_events = qw(comment_create comment_update comment_delete);
+
+sub new {
+    my $class = shift;
+    my $self  = $class->SUPER::new(@_);
+    $self->{app}          = undef;
+    $self->{config}       = undef;
+    $self->{client}       = undef;
+    $self->{channel}      = undef;
+    $self->{reconnecting} = 0;
+    return $self;
+}
+
+sub register {
+    my $self = shift;
+    $self->{app}    = shift;
+    $self->{config} = $self->{app}->config;
+    my $ioloop = Mojo::IOLoop->singleton;
+
+    $self->connect();
+
+    # register for events
+    for my $e (@job_events) {
+        $ioloop->on("openqa_$e" => sub { shift; $self->on_job_event(@_) });
+    }
+    for my $e (@comment_events) {
+        $ioloop->on("openqa_$e" => sub { shift; $self->on_comment_event(@_) });
+    }
+}
+
+sub reconnect {
+    my $self = shift;
+
+    return if $self->{reconnecting};
+    $self->{reconnecting} = 1;
+    Mojo::IOLoop->timer(
+        $self->{config}->{reconnect_timeout} => sub {
+            $self->{reconnecting} = 0;
+            $self->connect();
+        });
+}
+
+sub connect {
+    my $self = shift;
+
+    OpenQA::Utils::log_debug("Connecting to AMQP server");
+    $self->{client} = Mojo::RabbitMQ::Client->new(url => $self->{config}->{amqp}{url});
```

Comment:
To make the rest of the code a little cleaner you could also do something like this.
```
my $client = $self->{client} = Mojo::RabbitMQ::Client->new(url => $self->{config}->{amqp}{url});
```
And then use `$client` in the current scope.